dive_core/common/emulate_pm4.cpp

```diff
@@ -929,6 +929,114 @@ bool EmulatePM4::AdvanceCb(const IMemoryManager &mem_manager,
                 return false;
         }
     }
+    else if ((type == Pm4Type::kType7) && (type7_header.opcode == CP_START_BIN))
+    {
+
+        PM4_CP_START_BIN packet;
+        DIVE_VERIFY(mem_manager.CopyMemory(&packet,
+                                           emu_state_ptr->m_submit_index,
+                                           emu_state_ptr->GetCurIb()->m_cur_va,
+                                           sizeof(PM4_CP_START_BIN)));
+
+        // The CP_START_BIN & CP_END_BIN are pm4s only availabe at a650+
+        // here is the layout:
+        //  |CP_START_BIN|
+        //  |Prefix_block_0|
+        //  |Prefix_block_1|
+        //  ...
+        //  |Prefix_block_n|
+        //  |Common_block|
+        //  |CP_END_BIN|
+        // All the prefix blocks are in a separate IB
+        // Prefix_block contains setup for each tile, like viewport/scissors with fixed block size
+        // Common_block contains drawcalls and resolve
+
+        // only add the 1st prefix block here
+        // after this ib, we jump back to the Common_block
+        // once we advance to the CP_END_BIN, we add the rest ibs
+        // it is done this way since we don't know the size of the Common_block at this point
```

Comment:
Do we not? The CP_START_BIN packet has a BODY_DWORDS field. Doesn't that field tell you how big the common block is?

Reply:
that filed is always 0 in both 2 test captures. :(

Reply:
Gotcha. Would be good to add that to your comment block as well, for whoever reads it next

Reply:
updated the comment as suggested